src/pages/index.js

```diff
@@ -32,29 +45,87 @@ editFormValidator.enableValidation();
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    job: inputValues.description,
+// Get User Info
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      job: res.about,
+    });
+    userInfo.setUserAvatar({ avatar: res.avatar });
+  })
+  .catch((err) => {
+    console.error(err);
   });
+
+// Update User Info
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.setButtonText("Saving...");
+  api
+    .updateUserInfo(inputValues.title, inputValues.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: inputValues.title,
+        job: inputValues.description,
+      });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setButtonText("Save");
+    });
   profileEditPopup.close();
 }
 
-profileEditButton.addEventListener("click", openEditProfileModal);
-
 const profileEditPopup = new PopupWithForm(
   { popupSelector: "#profile-edit-modal" },
   handleProfileEditSubmit
 );
 profileEditPopup.setEventListeners();
 
-function openEditProfileModal() {
-  const displayInfo = userInfo.getUserInfo();
-  profileTitleInput.value = displayInfo.name;
-  profileDescriptionInput.value = displayInfo.job;
+profileEditButton.addEventListener("click", () => {
   profileEditPopup.open();
+});
+
+// Update User Avatar
+const avatarFormValidator = new FormValidator(config, profileImageForm);
+avatarFormValidator.enableValidation();
+
+const profileImagePopup = new PopupWithForm(
+  {
+    popupSelector: "#profile-image-edit-modal",
+  },
+  handleProfilePicSubmit
+);
+profileImagePopup.setEventListeners();
+
+profileImageButton.addEventListener("click", () => {
+  profileImagePopup.open();
+});
+
+function handleProfilePicSubmit(inputValues) {
+  profileImagePopup.setButtonText("Saving...");
+  api
+    .updateAvatar(inputValues.description)
+    .then(() => {
+      userInfo.setUserAvatar({ avatar: inputValues.description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileImagePopup.setButtonText("Save");
+    });
+  profileImagePopup.close();
+  profileImageForm.reset();
+  avatarFormValidator.resetValidation();
```

Comment:
You need to **reset** the form only when you **open** it or better in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more